vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -513,12 +605,20 @@ and
             match VsRunningDocumentTable.FindDocumentWithoutLocking(package.RunningDocumentTable,filename) with
             | Some (hier, _) ->
                 match hier with
-                | :? IProvideProjectSite as siteProvider when not (IsScript(filename)) -> 
+                | :? IProvideProjectSite as siteProvider when not (IsScript(filename)) ->
                     this.SetupProjectFile(siteProvider, this.Workspace, "SetupNewTextView")
-                | _ -> 
+                | _ when not (IsScript(filename)) ->
```

Comment:
When and why would ``hier`` not implement ``IProvideProjectSite``?  Please document this, it is some case that is not intuitive.

Reply:
CPS loaded projects don't support IProvideProjectSite.  Which is why we have to do this wrapping thingy.